helix-view/src/editor.rs

```diff
@@ -78,6 +72,25 @@ impl Editor {
         self.status_msg = Some((error, Severity::Error));
     }
 
+    pub fn set_theme(&mut self, theme: Theme) {
+        let config_loader = self.config_loader.clone();
+        let (_, doc) = self.current();
+        doc.detect_language(Some(&theme), config_loader);
```

Comment:
This really needs to be done in _all_ documents, since the list of scopes might have changed. We also don't need to trash the whole syntax tree, simply update the highlight configuration on it I think.